### DISCUSSION TITLE
How to read an event when user close modal?

### DISCUSSION THREAD

**Discussion by flameblue59** [wire-elements/modal#465] [Q&A]
I would like to dispatch an event when a user close the modal both click from the outside of the modal or from the close button in the modal. There is no option about onModalClose() that firing an event. Any ideas about this?

things I would like to do is to reset the page.

Here is the scenario

Try using in the modal

`use WithPagination;`

Then just go to other page. Then you can return by closing modal. The root url is totally changed. Here is the example:

http://localhost/socialmedia/instagram?page=5

So, where is the error? Now you can open another modal which the data less then the last modal opened. Then it would return empty. Or try to refresh the page. If there is a paginate inside it then it would also empty.

**Comment by iRaziul** [comment]
there is a method named `closeModalWithEvents` that can be used for your case.
